server/routes/prisonerChangeIncentiveLevelDetails.test.ts

```diff
@@ -0,0 +1,324 @@
+jest.mock('../services/userService')
+jest.mock('../data/hmppsAuthClient')
+jest.mock('../data/offenderSearch')
+jest.mock('../data/nomisUserRolesApi')
+
+let app: Express
+
+const bookingId = 12345
+const prisonerNumber = 'A8083DY'
+
+const tokenWithMissingRole = createUserToken([])
+const tokenWithNecessaryRole = createUserToken(['ROLE_MAINTAIN_IEP'])
+
+const incentiveSummaryForBooking: IncentiveSummaryForBookingWithDetails = {
+  bookingId,
+  iepDate: '2017-08-15',
+  iepTime: '2017-08-15T16:04:35',
+  iepLevel: 'Standard',
+  daysSinceReview: 1868,
+  nextReviewDate: '2018-08-15',
+  iepDetails: [
+    {
+      bookingId,
+      iepDate: '2017-08-15',
+      iepTime: '2017-08-15T16:04:35',
+      agencyId: 'MDI',
+      iepLevel: 'Standard',
+      userId: 'NOMIS_USER',
+      comments: 'STANDARD_NOMIS_USER_COMMENT',
+    },
+    {
+      bookingId,
+      iepDate: '2017-08-10',
+      iepTime: '2017-08-10T16:04:35',
+      agencyId: 'LEI',
+      iepLevel: 'Basic',
+      userId: 'SYSTEM_USER',
+      comments: 'BASIC_SYSTEM_USER_COMMENT',
+    },
+    {
+      bookingId,
+      iepDate: '2017-08-07',
+      iepTime: '2017-08-07T16:04:35',
+      agencyId: 'MDI',
+      iepLevel: 'Enhanced',
+      userId: 'UNKNOWN_USER',
+      comments: 'ENHANCED_UNKNOWN_USER_COMMENT',
+    },
+  ],
+}
+
+const prisonerDetails: Offender = {
+  offenderNo: prisonerNumber,
+  agencyId: 'MDI',
+  bookingId,
+  firstName: 'John',
+  lastName: 'Smith',
+  assignedLivingUnit: {
+    agencyId: 'MDI',
+    locationId: 1,
+    description: '123',
+    agencyName: '123',
+  },
+}
+
+const staffDetails: Staff = {
+  firstName: '123',
+  lastName: '123',
+  staffId: 123,
+  username: 'SYSTEM_USER',
+  activeCaseLoadId: '123',
+  active: true,
+}
+
+const offenderDetails: OffenderSearchResult = {
+  bookingId,
+  prisonerNumber,
+  firstName: 'John',
+  lastName: 'Smith',
+  prisonId: 'MDI',
+  prisonName: 'Moorland',
+  cellLocation: '123',
+}
+
+const userCaseload: UserCaseload = {
+  activeCaseload: {
+    id: 'MDI',
+    name: 'MDI',
+  },
+  caseloads: [
+    {
+      id: 'MDI',
+      name: 'MDI',
+    },
+  ],
+}
+
+const prisonApi = PrisonApi.prototype as jest.Mocked<PrisonApi>
+const offenderSearch = OffenderSearchClient.prototype as jest.Mocked<OffenderSearchClient>
+const incentivesApi = IncentivesApi.prototype as jest.Mocked<IncentivesApi>
+const nomisUserRolesApi = NomisUserRolesApi.prototype as jest.Mocked<NomisUserRolesApi>
+beforeEach(() => {
+  prisonApi.getPrisonerDetails.mockResolvedValue(prisonerDetails)
+  prisonApi.getStaffDetails.mockResolvedValue(staffDetails)
+  prisonApi.getAgency.mockImplementation(agencyId => {
+    if (agencyId === 'MDI') {
+      return Promise.resolve({
+        agencyId: 'MDI',
+        description: '123',
+        agencyType: '123',
+        active: true,
+      })
+    }
+    return Promise.resolve({
+      agencyId: 'LEI',
+      description: '123',
+      agencyType: '123',
+      active: true,
+    })
+  })
```

Comment:
from what i can tell, `prisonApi.getStaffDetails` and `prisonApi.getAgency` are not called by this page

Reply:
You would be correct!